source/dofs/dof_tools_constraints.cc

```diff
@@ -2098,10 +2098,13 @@ namespace DoFTools
         const unsigned int n_dofs_per_face =
           face_1->get_fe(face_1->nth_active_fe_index(0)).dofs_per_face;
 
-        Assert(matrix.m() == 0 || matrix.m() == n_dofs_per_face ||
-               matrix.m() == (int)spacedim,
-               ExcMessage ("matrix must have either size 0 or spacedim or the "
-                           "size must be equal to the # of DoFs on the face"));
+        Assert(matrix.m() == 0
+               || (first_vector_components.empty() && matrix.m() == n_dofs_per_face)
+               || (!first_vector_components.empty() && matrix.m() == (int)spacedim),
+               ExcMessage ("matrix must have either size 0 or spacedim "
+                           "(first_vector_components needed)"
+                           "or the size must be equal to the # of DoFs on the face"
+                           "(first_vector_components empty)"));
```

Comment:
Can you make this error message (and the other one you change below) into a complete sentence, i.e., starting with a capital letter and ending with a period? You also need a space at the end of the second through fourth line.